app/modules/entities/components/layouts/relative_entities_list.svelte

```diff
@@ -1,69 +1,107 @@
 <script>
+  import { i18n } from '#user/lib/i18n'
+  import { onChange } from '#lib/svelte/svelte'
+  import { isNonEmptyArray } from '#lib/boolean_tests'
+  import { forceArray } from '#lib/utils'
+  import { uniq, indexBy } from 'underscore'
   import Spinner from '#components/spinner.svelte'
   import { getEntitiesAttributesByUris, getReverseClaims, serializeEntity } from '#entities/lib/entities'
   import { addEntitiesImages } from '#entities/lib/types/work_alt'
   import Flash from '#lib/components/flash.svelte'
-  import ImagesCollage from '#components/images_collage.svelte'
-  import { forceArray, loadInternalLink } from '#lib/utils'
-  import { uniq } from 'underscore'
+  import SectionLabel from '#entities/components/layouts/section_label.svelte'
+  import RelativeEntityLayout from '#entities/components/layouts/relative_entity_layout.svelte'
 
   export let entity, property, label, claims
 
-  let flash, entities
+  let flash
+  let uris
 
-  const { uri, image } = entity
+  const { uri } = entity
 
   async function getUris () {
-    let uris
+    let allUris
     if (claims) {
-      uris = claims
+      allUris = claims
     } else {
       const properties = forceArray(property)
-      uris = await Promise.all(properties.map(property => {
+      allUris = await Promise.all(properties.map(property => {
         return getReverseClaims(property, uri)
       }))
     }
-    return uniq(uris.flat())
+    uris = uniq(allUris.flat())
   }
 
   const waiting = getUris()
-    .then(async uris => {
-      const res = await getEntitiesAttributesByUris({
-        uris,
-        // TODO: also request 'popularity' to be able to use it to sort the entities
-        attributes: [ 'type', 'labels', 'image' ],
-        lang: app.user.lang,
-      })
-      entities = Object.values(res.entities).map(serializeEntity)
-      await addEntitiesImages(entities)
+    .then(getAndSerializeEntities)
+
+  async function getAndSerializeEntities (uris) {
+    return getEntitiesAttributesByUris({
+      uris,
+      // TODO: also request 'popularity' to be able to use it to sort the entities
+      attributes: [ 'type', 'labels', 'image' ],
+      lang: app.user.lang,
     })
-    .catch(err => flash = err)
+      .then(async res => {
+        const entities = Object.values(res.entities).map(serializeEntity)
+        await addEntitiesImages(entities)
+        return indexBy(entities, 'uri')
+      })
+      .catch(err => flash = err)
+  }
+
+  let entitiesByUris = []
+  let loadMore, displayedUris
+
+  async function getMissingEntities () {
+    let missingUris = displayedUris.filter(p => !entitiesByUris[p])
+    if (isNonEmptyArray(missingUris)) {
+      loadMore = true
+      const missingEntities = await getAndSerializeEntities(missingUris)
+      loadMore = false
```

Comment:
If I'm not mistaken, if `getAndSerializeEntities` throws an error then `loadMore` will stay true. That can be avoided by storing the promise rather than a boolean
```suggestion
      loadingMore = getAndSerializeEntities(missingUris)
      const missingEntities = await loadingMore
```
which can then be `{#await}`ed

Reply:
done in 16a8e22060f97c5c904f498aaee1aeed37c392da